common/autoware_universe_utils/include/autoware/universe_utils/ros/static_transform_buffer.hpp

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE__UNIVERSE_UTILS__ROS__STATIC_TRANSFORM_BUFFER_HPP_
+#define AUTOWARE__UNIVERSE_UTILS__ROS__STATIC_TRANSFORM_BUFFER_HPP_
+
+#include "autoware/universe_utils/ros/transform_listener.hpp"
+
+#include <eigen3/Eigen/Core>
+#include <pcl_ros/transforms.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <sensor_msgs/point_cloud2_iterator.hpp>
+
+#include <pcl_conversions/pcl_conversions.h>
+
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+namespace std
+{
+template <>
+struct hash<std::pair<std::string, std::string>>
+{
+  size_t operator()(const std::pair<std::string, std::string> & p) const
+  {
+    size_t h1 = std::hash<std::string>{}(p.first);
+    size_t h2 = std::hash<std::string>{}(p.second);
+    return h1 ^ (h2 << 1u);
+  }
+};
+}  // namespace std
+
+namespace autoware::universe_utils
+{
+using std::chrono_literals::operator""ms;
+using Key = std::pair<std::string, std::string>;
+struct PairEqual
+{
+  bool operator()(const Key & p1, const Key & p2) const
+  {
+    return p1.first == p2.first && p1.second == p2.second;
+  }
+};
+using TFMap = std::unordered_map<Key, Eigen::Matrix4f, std::hash<Key>, PairEqual>;
+
+class StaticTransformBuffer
+{
+public:
+  StaticTransformBuffer() = default;
+
+  bool getTransform(
+    rclcpp::Node * node, const std::string & target_frame, const std::string & source_frame,
+    Eigen::Matrix4f & eigen_transform)
+  {
+    auto key = std::make_pair(target_frame, source_frame);
+    auto key_inv = std::make_pair(source_frame, target_frame);
+
+    // Check if the transform is already in the buffer
+    auto it = buffer_.find(key);
+    if (it != buffer_.end()) {
+      eigen_transform = it->second;
+      return true;
+    }
+
+    // Check if the inverse transform is already in the buffer
+    auto it_inv = buffer_.find(key_inv);
+    if (it_inv != buffer_.end()) {
+      eigen_transform = it_inv->second.inverse();
+      buffer_[key] = eigen_transform;
+      return true;
+    }
+
+    // Check if transform is needed
+    if (target_frame == source_frame) {
+      eigen_transform = Eigen::Matrix4f::Identity();
+      buffer_[key] = eigen_transform;
+      return true;
+    }
```

Comment:
Wouldn't it be more efficient to have this code at the beginning of the function?

Reply:
@TakaHoribe thank you for your review.
We loosing negligible time here on first request, but in next request we will have transformation in our buffer already (first condition). It is about nanoseconds. I would rather to have all history of TF requests in buffer.
* Regarding autoware_utils, we already have `autoware:universe_utils:TransformListener` and this PR brings similar functionalities, thus I put it alongside.
* We discussed dynamic calibration internally and came up with idea of adding service server for such a feature. Request will force TFs update. What do you think about it? We can also investigate custom mechanics for TF listener: default listener under the hood is just two callbacks to /tf and /tf_static. We could check where is transformation and perform suitable operations: for static TFs same as this PR, for dynamic TFs listen as default. It is just idea at it will need deeper investigation for possible drawbacks etc. Applying this for all the Autoware might free some CPU load.

Reply:
Thank you for your reply. We need to investigate the easy way for dynamic calibration, but so far the approach in this PR and your comment are reasonable. 

> Request will force TFs update. What do you think about it?
You mean, when this instance is created, a server automatically starts, and if dynamic calibration is being performed, the executing application will send a request to that service, triggering a force update? I think this is one of a good idea :+1: